### PR TITLE
azure front door fallback

### PR DESCRIPTION
**Pull request by emilhotkowski**
### Description

We now have fallback for frontdoor service. We directly call cluster in case of failure of FrontDoor service failure. I am using here [opossum](https://www.npmjs.com/package/opossum) for CircuitBreaker pattern to fallback for different url.
Circuit Breaker gives us : 
- fail fast
- ootb fallback mechanism

### Tested

Unit test was added.

### Backwards compatibility

It is compatible

### Extra info
@codyborn we still need to add fallbackUrl to the configuration. Should this be done after merge or before?

### CODE REVIEW

File: packages/phone-number-privacy/combiner/src/signing/get-threshold-signature.ts [hunk 1]
```diff
@@ -310,8 +312,25 @@ function requestSignature(
   controller: AbortController,
   logger: Logger
 ): Promise<FetchResponse> {
-  logger.debug({ signer: service.url }, `Requesting partial sig`)
-  const url = service.url + PARTIAL_SIGN_MESSAGE_ENDPOINT
+  const signerCircuitBreaker = new CircuitBreaker(parametrizedSignatureRequest)
+  signerCircuitBreaker.fallback(() =>
+    parametrizedSignatureRequest(service.fallbackUrl, request, controller, logger)
```
Comment (codyborn) [inline packages/phone-number-privacy/combiner/src/signing/get-threshold-signature.ts]: We should allow `fallbackUrl` to not be specified for some signers.

File: packages/phone-number-privacy/combiner/src/signing/get-threshold-signature.ts [hunk 2]
```diff
@@ -310,8 +312,25 @@ function requestSignature(
   controller: AbortController,
   logger: Logger
 ): Promise<FetchResponse> {
-  logger.debug({ signer: service.url }, `Requesting partial sig`)
-  const url = service.url + PARTIAL_SIGN_MESSAGE_ENDPOINT
+  const signerCircuitBreaker = new CircuitBreaker(parametrizedSignatureRequest)
```
Comment (codyborn) [inline packages/phone-number-privacy/combiner/src/signing/get-threshold-signature.ts]: It looks like we can also add a timeout here. We have a global combiner timeout defined as `timeoutMilliSeconds` in the config, but perhaps we should define a new one that's for each request. WDYT?

**Comment by codyborn** [comment]
Nice! 

For tests, we can add them to the [index.test.ts](https://github.com/celo-org/celo-monorepo/blob/master/packages/phone-number-privacy/combiner/test/index.test.ts) which tests `getBlindedMessageSig`.

We should update the config before deploying; however we should also make the fallback URL optional since not all signers will have one.

File: packages/phone-number-privacy/combiner/src/signing/get-threshold-signature.ts (hunk 1 above)
Comment (emilhotkowski) [reply packages/phone-number-privacy/combiner/src/signing/get-threshold-signature.ts]: Added this as an optional parameter.

File: packages/phone-number-privacy/combiner/src/signing/get-threshold-signature.ts (hunk 2 above)
Comment (emilhotkowski) [reply packages/phone-number-privacy/combiner/src/signing/get-threshold-signature.ts]: Added

**Comment by emilhotkowski** [comment]
@codyborn I've upgraded the code so it handles one circuit breaker per signer. Do you have any idea why do tests fail? I am not sure if this is on me.

**Review by codyborn — APPROVED** [review]
LGTM! Thanks

**Review by alecps — APPROVED** [review]
Looks good. We should test this e2e in staging before we deploy to alfajores or mainnet

**Pull request merged**